File: singly_Linked_list.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None
class header:
    def __init__(self):
        self.head = None
class operations:
    def insAtBeg(self,d):
        temp = Node(d)
        temp.next = h.head
        h.head = temp
    def insAtEnd(self,d):
        temp = Node(d)
        trav = h.head
        if(h.head == None):
            insAtBeg(self,d)
        else:
            while(trav.next != None):
                trav = trav.next
            trav.next = temp
    def insAtPos(self,pos,d):
        temp = Node(d)
        if(pos == 1):
            insAtBeg(self,d)
        else:
            c= 2
            trav = h.head
            while(c != pos):
               c+=1
               trav = trav.next
            temp.next = trav.next
            trav.next = temp
    def delAtBeg(self):
        temp = h.head
        h.head = temp.next
        del temp
    def delAtEnd(self):
        trav = h.head
        while(trav.next.next != None):
            trav = trav.next
        temp = trav.next
        trav.next = None
        del temp
    def delAtPos(self,pos):
        if(pos == 1):
            delAtBeg(self)
        else:
            c= 2
            trav = h.head
            while(c!=pos):
                c+=1
                trav = trav.next
            temp = trav.next
            trav.next = temp.next
            del temp
    def search(self,pos):
        if(pos ==1):
            print(h.head.data)
        else:
            c= 2
            trav = h.head
            while(c != pos):
                trav = trav.next
            print(trav.data)
    def print(self):
        trav = h.head
        while(trav != None):
            print(trav.data,end =" ")
            trav = trav.next
            
                



h = header()
op = operations()
print("Linked list implementation")
choice = int(input("1. Insert at beginning \n2.Insert at end\n3.Insert at pos\n4.Selection at beginning\n5.Deletion at end\n6.Deletion at position\n7.Search\n8.Print elements\n9.Exit\n"))
while(choice != 9):
    if(choice == 1):
        data = int(input("Enter the data to enter"))
        op.insAtBeg(data)
    elif(choice == 2):
        data = int(input("Enter the data to enter"))
        op.insAtEnd(data)
    elif(choice == 3):
        data = int(input("Enter the data to enter"))
        pos =  int(input("Enter the pos to enter"))
        op.insAtPos(pos,data)
    elif(choice == 4):
        op.delAtBeg()
    elif(choice == 5):
        op.delAtEnd()
    elif(choice == 6):
        pos = int(input("Enter the pos to delete"))
        op.delAtPos(pos)
    elif(choice == 7):
        pos = int(input("Enter the pos to search"))
        op.search(pos)
    elif(choice == 8):
        op.print()
    choice = int(input("Enter your choice"))

File: Singly_linked_list.java
public class Node {
    int data;
    Node next;
  
    Node(int d)
    {
        data = d;
        next = null;
    }

}
public class SinglyLinkedList {
    Node head = null;
       
    static int data,pos;
     private void InsertAtEnd(int data) {
        if(head == null)
        {
            InsertAtBeg(data);
        }
        else
        {
            Node node = new Node(data);
            Node temp = head;
            while(temp.next != null)
                temp = temp.next;
            temp.next = node;
        }
    }

    private void InsertAtBeg(int data) {
        if(head == null)
        {
            Node node = new Node(data);
            head = node;
        }
        else
        {
            Node node = new Node(data);
            //Node temp = Node.head;
            node.next = head;
            head = node;
        }
    }

    private void InsertAtPos(int data, int pos) {
        if(pos == 1)
            InsertAtBeg(data);
        else
        {
            Node node = new Node(data);
            int count = 1;
            Node temp = head;
            while(temp.next!=null && count<pos)
            {
                temp = temp.next;
                count++;
            }
            Node chnge = temp.next;
            temp.next = node;
            node.next = chnge;
        }
    }

    private void DelAtBeg() {
        if(head != null)
        {
            Node temp = head;
            head = head.next;
        }
    }

    private void DelAtEnd() {
        if(head.next == null)
        {
            head = null;
        }
        else
        {
            Node temp = head;
            while(temp.next.next != null)
                temp = temp.next;
            temp.next = null;
        }
    }

    private void DelAtPos(int pos) {
        int count = 1;
        Node trav = head;
        while(count< pos-2)
            trav = trav.next;
        Node temp = trav.next.next;
        trav.next = temp;
    }
    private void Display() {
        Node trav = head;
        while(trav != null)
        {
            System.out.println(trav.data);
            trav = trav.next;
        }
    }

    private void search(int data) {
        Node trav = head;
        pos = 1;
        int flag =  0;
        while(trav != null)
        {
            if(trav.data == data)
            {   
                flag = 1;
                System.out.println("The data is present in position "+ pos);
            }
            trav = trav.next;
            pos++;
        }
        if(flag == 0)
            System.out.println("The data is not present");
        
    }

    public static void main(String[] args) {
        SinglyLinkedList sl = new SinglyLinkedList();
        Scanner scan  =  new Scanner(System.in);
        System.out.println("Enter your choice");
        System.out.println("1.Insert at beginning \n 2.Insert at end \n 3.Insert at position \n 4.Delete at beginning \n 5.Delete at end \n 6.Delete at position \n 7.Search \n 8.Displayt \n9.Exit");
        int choice = scan.nextInt();
        while(choice != 9)
        {
            if(choice == 1)
            {
                System.out.println("Enter the data to insert");
                data = scan.nextInt();
                sl.InsertAtBeg(data); 
                 sl.Display();
            }
            if(choice == 2)
            {
                System.out.println("Enter the data to insert");
                data = scan.nextInt();
                sl.InsertAtEnd(data);
                 sl.Display();
            }
            if(choice == 3)
            {
                System.out.println("Enter the data to insert");
                data = scan.nextInt();
                System.out.println("Enter the position to enter data");
                pos= scan.nextInt();
                sl.InsertAtPos(data,pos);
                 sl.Display();
            }
            if(choice == 4)
            {
                sl.DelAtBeg();
                 sl.Display();
            }
            if(choice == 5)
            {
                sl.DelAtEnd();
                 sl.Display();
            }
            if(choice == 6)
            {
                System.out.println("Enter the position to enter data");
                pos= scan.nextInt();
                sl.DelAtPos(pos);
                 sl.Display();
            }
            if(choice == 7)
            {
                System.out.print("Enter the data to print");
                data = scan.nextInt();
                sl.search(data);
            }
            if(choice == 8)
            {
                sl.Display();
            }
            System.out.println("1.Insert at beginning \n 2.Insert at end \n 3.Insert at position \n 4.Delete at beginning \n 5.Delete at end \n 6.Delete at position \n 7.Search \n 8.Display \n 9.Exit");
            choice= scan.nextInt();
        }
    }
    
}
